### DISCUSSION TITLE
Application upgrade issues

### DISCUSSION THREAD

**Discussion by claybrooks** [wixtoolset/issues#7863] [Questions]
I'm running in to an issue with Wix installers during an upgrade path.  My new installer is not detecting the previously installed version. 
 Some things about the setup that may be important:

- Old version of software installed with Wix v3
- New version of software installed with Wix v4
- Rebranding of some application metadata and folder locations
&nbsp;&nbsp;&nbsp;&nbsp; - ProductName changed
&nbsp;&nbsp;&nbsp;&nbsp; - Manufacturer changed
&nbsp;&nbsp;&nbsp;&nbsp; - Default folder name within Program Files (x86) changed
- Minor version of our application bumped (1.1 -> 1.2)
- UpgradeCode **did not** change

In some aspects, Wix is not detecting the previous version when launching the new msi
- Running "msiexec /f setup.msi" (repair) reports an error message saying no version of the product is installed
- No landing page for repair/change/remove like I would expect during normal application upgrade
- Feature selection tree does not use the previous install selections, it uses the defaults we have setup in our codebase.

But in other aspects, Wix seems to be upgrading fine
 - After new install, all files in the old install location are gone
 - The old application name is not listed in add/remove programs

Here is a condensed Product.wxs.  I've included what I think is relevant for dealing with upgrades.
```
<Wix>
    <Package Manufacturer="NewCompany" Name="New Name" UpgradeCode="Same as it's always been" Version="1.2" InstallerVersion="405>
        <MajorUpgrade DowngradeErrorMessage="..." AllowDowngrades="no" AllowSameVersionUpgrades="yes" Schedule="afterInstallValidate" IgnoreRemovalFailure="yes" />
   </Package>
</Wix>
```

I'm happy to share more details about the setup if necessary.

**I was under the impression the UpgradeCode is the key for windows to know what the application is, despite the version and other metadata.  Is the rebranding causing issues with windows detecting a previous install?**

**Comment by Rparker19** [comment]
@claybrooks did you ever find a solution for this problem? I am running into the same issue with some software that was upgraded from Wix v3 to Wix v5 and haven't been able to find an answer anywhere

**Reply by claybrooks** [reply]
No, unfortunately I did not :(

**Reply by wmanning** [reply]
Correct, UpgradeCode is what MSI looks for when determining related msi's to upgrade.  
You have MajorUpgrade Schedule attribute set to "afterInstallValidate".  If the installation fails that will mean both products are not installed: https://docs.firegiant.com/wix3/xsd/wix/majorupgrade/ I use "afterInstallInitialize" so at least the old product is left installed on failure.  

Everything you listed seems OK.

Do you see anything in the MSI log about an error?  That would be the first thing to check.

There is a tool called [Orka](https://learn.microsoft.com/en-us/windows/win32/msi/orca-exe) that's in the Windows SDK.  The MSI file is a database of sorts and with this tool you allows you to view the database contents.  Maybe try building the previous MSI source with Wix4 and compare the two MSIs using Orca to see if there is a difference.

**Reply by Rparker19** [reply]
The problem for us ended up being that we were checking for the 
`DetectBeginEventArgs e.RegistrationType == RegistrationType.InProgress`, when in fact to detect a previous install, this should be `e.RegistrationType == RegistrationType.Full`.
